### PR TITLE
fix(core): manage unrecongnized langage

### PR DESCRIPTION
**Pull request by stonebuzz**
'transliterator_transliterate' maybe incompatible with some langage (cambodian) and return an empty string

We get an error from inventory with this software (for example)
```json
         {
            "arch": "neutral",
            "folder": "C:\\Program Files\\WindowsApps\\Microsoft.LanguageExperiencePackkm-KH_19038.0.1.0_neutral__8wekyb3d8bbwe",
            "from": "uwp",
            "install_date": "2022-01-01",
            "name": "ភាសាខ្មែរ កញ្ចប់បទពិសោធន៍ផ្ទៃក្នុង",
            "publisher": "Microsoft Corporation",
            "system_category": "store",
            "version": "19038.0.1.0"
         },
```

1. If ```transliterator_transliterate``` return an empty string, reuses original string
2. Remove ```preg_replace``` on ```latin``` char (for no latin char).
3. Add TU to be sure ```slugify``` return a string (```!=``` ```nok_```)

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Comment by trasher** [comment]
Please add a test on original issue #2946.

**Comment by trasher** [comment]
BTW, it would be useful to test if slugify returns expected value when original string starts with a number

File: src/Toolbox.php [hunk 1]
```diff
@@ -2622,16 +2622,20 @@ public static function in_array_recursive($needle, $haystack, $strict = false)
...
      * Slugify
      *
-     * @param string $string String to slugify
+     * @param string $str String to slugify
      * @param string $prefix Prefix to use (anchors cannot begin with a number)
      *
      * @return string
      */
-    public static function slugify($string, $prefix = 'slug_')
+    public static function slugify($str, $prefix = 'slug_')
```
Comment (cedric-anne) [inline src/Toolbox.php:2630]: You should probably add a `bool $allow_special_chars = false` parameter to preserve previous behaviour by default. Indeed, special chars may be an issue when slug is used for an HTML id attribute or an URL.

**Comment by stonebuzz** [comment]
> Please add a test on original issue #2946.

done

**Comment by cedric-anne** [comment]
This will make following test obsolete, I guess: https://github.com/glpi-project/glpi/blob/10.0/bugfixes/src/Inventory/Asset/Software.php#L104-L113

Maybe, instead of using slugify to build deduplication key we could use sha1.

**Review by orthagh — APPROVED** [review]

**Comment by stonebuzz** [comment]
> This will make following test obsolete, I guess: https://github.com/glpi-project/glpi/blob/10.0/bugfixes/src/Inventory/Asset/Software.php#L104-L113
> 
> Maybe, instead of using slugify to build deduplication key we could use sha1.

Indeed, this check is obsolete

Need to add some code to get ```comments``` as ```name``` if ```name``` is empty and if ```guid``` is empty too 

Like this (```guid``` is missing): 

```json
         {
            "arch": "neutral",
            "comments": "Microsoft Edge Browser",
            "folder": "C:\\Program Files\\WindowsApps\\Microsoft.MicrosoftEdge.Stable_107.0.1418.56_neutral__8wekyb3d8bbwe",
            "from": "uwp",
            "install_date": "2022-11-23",
            "name": "",
            "publisher": "Microsoft Corporation",
            "system_category": "developer",
            "version": "107.0.1418.56"
         }
```

Else GLPI trigger PHP error

```shell
[2022-11-23 14:13:58] glpiphplog.ERROR:   *** PHP User Error (256): Incorrect integer value: 'Microsoft Corporation' for column `10bugfixes_ruless`.`glpi_softwares`.`manufacturers_id` at row 1 in /home/stanislas/TECLIB/DEV/GLPI/10.0-bugfixes/src/DBmysql.php at line 1969
  Backtrace :
  src/DBmysql.php:1969                               trigger_error()
  src/Inventory/Asset/Software.php:692               DBmysql->executeStatement()
  src/Inventory/Asset/Software.php:422               Glpi\Inventory\Asset\Software->storeSoftware()
  src/Inventory/Asset/MainAsset.php:862              Glpi\Inventory\Asset\Software->handle()
  src/Inventory/Asset/MainAsset.php:782              Glpi\Inventory\Asset\MainAsset->handleAssets()
  src/RuleImportAsset.php:969                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src/Rule.php:1522                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1591                        Rule->process()
  src/Inventory/Asset/MainAsset.php:564              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:706                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:341                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:90                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:332    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:244    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  src/Inventory/Conf.php:254                         Glpi\Agent\Communication\AbstractRequest->handleRequest()
  src/Inventory/Conf.php:195                         Glpi\Inventory\Conf->importContentFile()
  front/inventory.conf.php:47                        Glpi\Inventory\Conf->importFile()
 
```
Because ```$val->name``` is not set and ```insert``` is made by this (and columns are moved one step)

```
$stmt->bind_param($stmt_types, ...$stmt_values);
$DB->executeStatement($stmt);
```

**Comment by cedric-anne** [comment]
#13390 is a better fix, but some added tests could be added to test suite.

**Comment by stonebuzz** [comment]
I close this one in favor of  https://github.com/glpi-project/glpi/pull/13390

**Pull request closed without merging**